server/server.go

```diff
@@ -53,38 +59,26 @@ func (cs *cribbageServer) Serve() {
 }
 
 func (cs *cribbageServer) ginPostCreateGame(c *gin.Context) {
-	var pIDs []model.PlayerID
-
-	pID := getPlayerID(c, `player1`)
-	if pID == model.InvalidPlayerID {
-		c.String(http.StatusBadRequest, `Needs player1`)
-		return
-	}
-	pIDs = append(pIDs, pID)
-
-	pID = getPlayerID(c, `player2`)
-	if pID == model.InvalidPlayerID {
-		c.String(http.StatusBadRequest, `Needs player2`)
+	var gameReq model.Game
+	err := c.ShouldBindJSON(&gameReq)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
 		return
 	}
-	pIDs = append(pIDs, pID)
-
-	pID = getPlayerID(c, `player3`)
-	if pID != model.InvalidPlayerID {
-		pIDs = append(pIDs, pID)
-	}
-
-	pID = getPlayerID(c, `player4`)
-	if pID != model.InvalidPlayerID {
-		pIDs = append(pIDs, pID)
+	pIDs := make([]model.PlayerID, len(gameReq.Players))
+	for i, p := range gameReq.Players {
+		if p.ID == model.InvalidPlayerID {
+			c.String(http.StatusBadRequest, `Invalid player ID at index %d`, i)
+			return
+		}
+		pIDs[i] = p.ID
 	}
 
 	if len(pIDs) < model.MinPlayerGame || len(pIDs) > model.MaxPlayerGame {
-		c.String(http.StatusBadRequest, `Invalid num players: %d`, len(pIDs))
+		c.String(http.StatusBadRequest, `Invalid num players: %d`, len(gameReq.Players))
 		return
 	}
-
-	g, err := createGameFromIDs(pIDs)
+	g, err := createGame(context.Background(), cs.dbService, pIDs)
```

Comment:
🔮 is there a context on the gin.Context we could use instead of `context.Background()`?

Reply:
```go
type Context struct {
    Request *http.Request
    Writer  ResponseWriter

    Params Params

    // Keys is a key/value pair exclusively for the context of each request.
    Keys map[string]interface{}

    // Errors is a list of errors attached to all the handlers/middlewares who used this context.
    Errors errorMsgs

    // Accepted defines a list of manually accepted formats for content negotiation.
    Accepted []string
    // contains filtered or unexported fields
}
```